builder/mixconfig.go

```diff
@@ -0,0 +1,194 @@
+package builder
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/go-ini/ini"
+)
+
+// MixConfig represents the config parameters found in the builder config file.
+type MixConfig struct {
+	// [Builder]
+	BundleDir  string
+	Cert       string
+	StateDir   string
+	VersionDir string
+	YumConf    string
+
+	// [swupd]
+	Bundle     string
+	ContentURL string
+	Format     string
+	VersionURL string
+
+	// [Server]
+	DebugInfoBanned string
+	DebugInfoLib    string
+	DebugInfoSrc    string
+
+	// [Mixer]
+	LocalBundleDir string
+	RepoDir        string
+	RPMDir         string
+}
+
+type configPair struct {
+	key   string
+	value *string
+}
+
+type configMapping struct {
+	section string
+	pairs   []configPair
+}
+
+func (config *MixConfig) getMapping() []configMapping {
```

Comment:
I see how mapping made sense if we are writing and reading, but I think you should just drop it and use a couple of switch statements to handle the cases. Reasons:

(a) in case of some options, we really want to have lists (bundle list) or numbers (versions, in the future), so we would need can do a preparse on the spot.
(b) we can add more validation as part of the reading of those values, i.e. never have an invalid value in config.

My suggestion is remove getMapping(), inline mapFromIni() into LoadConfig. To avoid long indentation, for each section you may want to call a different function to parse that section or something similar.

Reply:
@cmarcelo See my other comment about wanting the ability to write, not just read, the config programmatically.

Reply:
@cmarcelo my original implementation (https://github.com/rchiossi/mixer-tools/commit/305d2c9c25e4484186629c13ae3c1010e851ef63) had a discrete parsing of all the values, which would make it easier to validate all the entries.

Reply:
I like that one better than the pairs of name / pointer to property.